handler/core/src/main/java/io/knotx/fragments/task/TaskManager.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * The code comes from https://github.com/tomaszmichalak/vertx-rx-map-reduce.
+ */
+package io.knotx.fragments.task;
+
+import io.knotx.fragments.api.Fragment;
+import io.knotx.fragments.engine.FragmentEventContext;
+import io.knotx.fragments.engine.Task;
+import io.knotx.fragments.handler.action.ActionProvider;
+import io.knotx.fragments.task.exception.GraphConfigurationException;
+import io.knotx.fragments.task.exception.TaskNotFoundException;
+import io.knotx.fragments.task.options.TaskOptions;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+
+public class TaskManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TaskManager.class);
+
+  private String taskKey;
+  private List<TaskProviderFactory> providersFactories;
+  private Map<String, TaskOptions> tasks;
+  private ActionProvider actionProvider;
+
+  public TaskManager(String taskKey, Map<String, TaskOptions> tasks,
+      ActionProvider actionProvider) {
+    this.taskKey = taskKey;
+    this.tasks = tasks;
+    this.actionProvider = actionProvider;
+    providersFactories = initProviders();
+  }
+
+  public Optional<Task> get(FragmentEventContext fragmentEventContext) {
```

Comment:
`get` -> `newInstance` ?

Why `Optional`? Shouldn't something BAD happen when the factory is not able to provide a product (like some exception is thrown)?

Reply:
I think Task Manager should not decide what to do when task is not configured. We should do it in more configurable way.